packages/core/src/locales/fi.ts

```diff
@@ -0,0 +1,197 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import type { $ZodTypeDef } from "../schemas.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; subject: string }> = {
+  string: { unit: "merkkiä", subject: "merkkijonon" },
+  file: { unit: "tavua", subject: "tiedoston" },
+  array: { unit: "alkiota", subject: "listan" },
+  set: { unit: "alkiota", subject: "joukon" },
+  number: { unit: "", subject: "luvun" },
+  bigint: { unit: "", subject: "suuren kokonaisluvun" },
+  int: { unit: "", subject: "kokonaisluvun" },
+  date: { unit: "", subject: "päivämäärän" },
+};
+
+function getSizing(origin: string): { unit: string; subject: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+const TypeNames: { [k in $ZodTypeDef["type"] | (string & {})]?: string } = {
+  string: "merkkijono",
+  number: "luku",
+  boolean: "totuusarvo",
+  bigint: "suuri kokonaisluku",
+  symbol: "symboli",
+  null: "tyhjä",
+  undefined: "määrittelemätön",
+  date: "päivämäärä",
+  object: "objekti",
+  file: "tiedosto",
+  array: "lista",
+  map: "hakemisto",
```

Comment:
IMO better would be `"tietue"`/`"tietueessa"`
> avain hakemistossa
vrt
> avain tietueessa

https://fi.wikipedia.org/wiki/Tietue

"hakemisto" ottaa mielestäni liikaa kantaa implementoivaan tietorakenteeseen.
Voisin myös kuvitella että tavallisille käyttäjille "hakemisto" tuo mieleen tiedostjärjestelmän kansiorakenteen tms. Ehkä... 😸